.github/workflows/ci.yml

```diff
@@ -57,7 +57,7 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - { java-version: 23, cache: 'restore', cleanup-node: true }
```

Comment:
Is this valid? I think maven-checks does not build everything so you cannot cache here

Reply:
even if it's partial cache, this will still work - the cache will be populated with the fetched .m2 dependencies. Trino isn't cached.

Reply:
We should only populate the cache in a single job that builds everything, including using all maven plugins. This should be the error prone job. Ideally it'll run at least a single test to also fetch surefire plugin, etc.

Reply:
There is no such job, it's better to have partial cache than no cache at all. Error prone does not run everything

Reply:
Also we don't have go-offline anymore so these concerns do not apply anymore. Missing dependencies will be fetched over already cached ones

Reply:
> it's better to have partial cache than no cache at all

This is not about having partial cache vs no cache at all.
But about overwriting good cache (from job which runs everything or almost everything) with poor cache (from job which runs almost nothing).

I am not convinced you are not introducing problem here.

Reply:
I am confused - there was nothing with `cache: 'true'` before?

Reply:
Oh - this was artifact of java version migration. So this change is fine.

Reply:
Yeah. Nothing was cached before